packages/plugin-interactive-tools/sources/colorizeUtils.ts

```diff
@@ -0,0 +1,57 @@
+import {Configuration, formatUtils, structUtils} from '@yarnpkg/core';
+import {diffWords}                               from 'diff';
+
+const SIMPLE_SEMVER = /^((?:[\^~]|>=?)?)([0-9]+)(\.[0-9]+)(\.[0-9]+)((?:-\S+)?)$/;
+
+export function colorizeVersionDiff(configuration: Configuration, from: string, to: string) {
+  if (from === to)
+    return to;
+
+  const parsedFrom = structUtils.parseRange(from);
+  const parsedTo = structUtils.parseRange(to);
+
+  const matchedFrom = parsedFrom.selector.match(SIMPLE_SEMVER);
+  const matchedTo = parsedTo.selector.match(SIMPLE_SEMVER);
+
+  if (!matchedFrom || !matchedTo)
+    return colorizeRawDiff(configuration, from, to);
+
+  const SEMVER_COLORS = [
+    `gray`, // modifier
+    `red`, // major
+    `yellow`, // minor
+    `green`, // patch
+    `magenta`, // prerelease
+  ];
+
+  let color: string | null = null;
+  let res = ``;
+
+  for (let t = 1; t <= SEMVER_COLORS.length; ++t) {
```

Comment:
I think we could refactor this part to `Array.prototype.reduce`, but in order to keep the change small, I think we'd better leave this refactor to another pr.